### ISSUE TITLE
TIP Cognito: Don't get stuck on the username

### ISSUE THREAD

**Issue by awyork** [boylesoftware/tomcat-oidcauth#56]
Cognito will send back a GUID in the User Name (Sub) field, so you might want to setup your provider as below:

`{
               name: 'Amazon Cognito',
               issuer: https://cognito-idp.us-east-1.amazonaws.com/us-east-1_XXXXXXX,
               clientId: xxxxxxxxxxxxxx,
               clientSecret: xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
               usernameClaim: cognito:username
},`

**Comment by ishan-kpit** [comment]
Were you able to configure Cognito correctly @awyork ?
Currently I'm getting this error
`org.bsworks.catalina.authenticator.oidc.BaseOpenIDConnectAuthenticator.processAuthResponse failed to authenticate the user in the realm`

**Comment by awyork** [comment]
Yes.  It's been awhile, but if I remember correctly, the library was looking at the wrong field for the username. In the standard username field was a guid, but there was another field that did have the user name.  Also, you  must still authenticate on the Tomcat side.  

I unfortunately lost most of this work in a disk crash.

**Comment by awyork** [comment]
I've had a chance to look at Cognito again.  The "default" username is the GUID assigned by Cognito.  You can either use the "preferred_username" field or add a custom "username" field.
